test/1.4/serialize/T_saved_dead_vtable.dml

```diff
@@ -0,0 +1,29 @@
+/*
+  © 2022 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+dml 1.4;
+
+device test;
+
+/// COMPILE-ONLY
+
+template t {
+    shared method m();
+}
+
+template u {
+    shared method n();
+}
+
```

Comment:
add a saved uninstantiated template? I suppose this use case is why you write `trait_instances.get(x, ())` instead of `trait_instances[x]`?

Reply:
Sure, that's reasonable.

Reply:
And _am I glad you pointed that out_, it revealed a bug.

Reply:
Bug was that `_tinit_...` functions were generated despite never being used, which then gcc complains about. Fixed by not marking traits as referenced ourselves, but instead relying on `ObjTraits.mark_referenced` to do it for us.